test/com/google/enterprise/adaptor/sharepoint/SharePointAdaptorTest.java

```diff
@@ -1532,12 +1534,74 @@ public void testGetDocContentSCVerifyGroupPush() throws Exception {
         new DocId("http://localhost:1/sites/SiteCollection"));
     GetContentsResponse response = new GetContentsResponse(baos);
     adaptor.getDocContent(request, response);
+    assertEquals(goldenGroups1, pusher.getGroupDefinitions());
 
-    assertEquals(goldenGroups1, pusher.getGroupDefinitions());    
-    pusher.reset();
```

Comment:
I thought the call to pusher.reset() was germane to this test, verifying that we always pushed all of the groups. It does, however, suggest a third test, which is covered by the RecordingDocIdPusher tests in the library, but might make this call to reset more clear: Push disjoint groups in the two calls, or at least non-subset groups (overlapping is OK), and verify that the result is the union of the two calls. For example: { g1, g2 } + { g2, g3 } = { g1, g2, g3 }

Reply:
Not using pusher.reset() since in SP, it replaces all groups and users for every getDocContent request. Modified existing test to { g1, g2 } + { g2, g3 } = { g2, g3 }. 

Removed testGetDocContentSCDeleteGroupPush().